### PR TITLE
[New Rule] Suspicious File Modification

### PR DESCRIPTION
**Pull request by Aegrah**
## Summary
Research related to this PR is conducted and documented in https://github.com/elastic/ia-trade-team/issues/374. This issue also contains the query + validation aspect.

This PR:
- Adds a rule that leverages FIM to detect file modification in directories/files that are commonly abused for persistence.

### CODE REVIEW

File: rules/linux/persistence_suspicious_file_modifications.toml [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+  // runtime control
+  "/etc/rc.common", "/etc/rc.local",
+  // init daemon
+  "/etc/init.d/*",
+  // passwd/sudoers/shadow
+  "/etc/passwd", "/etc/shadow", "/etc/sudoers", "/etc/sudoers.d/*",
+  // Systemd udevd
+  "/lib/udev/*", "/etc/udev/rules.d/*", "/usr/lib/udev/rules.d/*", "/run/udev/rules.d/*", 
+  // XDG/KDE autostart entries
+  "/home/*/.config/autostart/*", "/root/.config/autostart/*", "/etc/xdg/autostart/*", "/usr/share/autostart/*",
```
Comment (w0rk3r) [inline rules/linux/persistence_suspicious_file_modifications.toml:87]:
```suggestion
  "/etc/cron.deny",  "/var/spool/anacron/*", "/var/spool/cron/atjobs/*",

  // systemd services & timers
  "/etc/systemd/system/*", "/usr/local/lib/systemd/system/*", "/lib/systemd/system/*",
  "/usr/lib/systemd/system/*", "/home/*/.config/systemd/user/*", "/home/*/.local/share/systemd/user/*",
  "/root/.config/systemd/user/*", "/root/.local/share/systemd/user/*",

  // LD_PRELOAD
  "/etc/ld.so.preload", "/etc/ld.so.conf.d/*", "/etc/ld.so.conf",

  // message-of-the-day (MOTD)
  "/etc/update-motd.d/*",

  // SSH
  "/home/*/.ssh/*", "/root/.ssh/*", "/etc/ssh/*",

  // system-wide shell configurations
  "/etc/profile", "/etc/profile.d/*", "/etc/bash.bashrc", "/etc/zsh/*", "/etc/csh.cshrc",
  "/etc/csh.login", "/etc/fish/config.fish", "/etc/ksh.kshrc",

  // root and user shell configurations
  "/home/*/.profile", "/home/*/.bashrc", "/home/*/.bash_login", "/home/*/.bash_logout",
  "/root/.profile", "/root/.bashrc", "/root/.bash_login", "/root/.bash_logout",
  "/home/*/.zprofile", "/home/*/.zshrc", "/root/.zprofile", "/root/.zshrc",
  "/home/*/.cshrc", "/home/*/.login", "/home/*/.logout", "/root/.cshrc", "/root/.login", "/root/.logout",
  "/home/*/.config/fish/config.fish", "/root/.config/fish/config.fish",
  "/home/*/.kshrc", "/root/.kshrc",

  // runtime control
  "/etc/rc.common", "/etc/rc.local",

  // init daemon
  "/etc/init.d/*",

  // passwd/sudoers/shadow
  "/etc/passwd", "/etc/shadow", "/etc/sudoers", "/etc/sudoers.d/*",

  // Systemd udevd
  "/lib/udev/*", "/etc/udev/rules.d/*", "/usr/lib/udev/rules.d/*", "/run/udev/rules.d/*",

  // XDG/KDE autostart entries
  "/home/*/.config/autostart/*", "/root/.config/autostart/*", "/etc/xdg/autostart/*", "/usr/share/autostart/*",
```
nit/optional, I think this improves the readability

File: rules/linux/persistence_suspicious_file_modifications.toml [hunk 2]
```diff
@@ -0,0 +1,181 @@
+[metadata]
+creation_date = "2024/06/03"
+maturity = "production"
```
Comment (w0rk3r) [inline rules/linux/persistence_suspicious_file_modifications.toml:3]:
```suggestion
maturity = "production"
integration = ["fim"]
```

**Review by w0rk3r — APPROVED** [review]

**Comment by terrancedejesus** [comment]
Nice! Couple of things:

- As we don't have any previous FIM rules, we should add the folder and dump this rule there to stay consistent.
- Pull integration manifests and schemas via `python -m detection_rules dev integrations build-manifests -o -i fim` then `python -m detection_rules dev integrations build-schemas -i fim` -> while the query validation unit tests are passing, that is because they are checking ECS which include all the file related fields.
- We should update [test_related_tags](https://github.com/elastic/detection-rules/blob/59b7e3bde4ae26ec03e52454905589abb496fb21/tests/test_all_rules.py#L291) and add FIM.
- Add `Data Source: File Integrity Monitoring` tag

**Comment by Aegrah** [comment]
Pushed the updates as requested. Is this correct? @terrancedejesus

File: rules/integrations/fim/persistence_suspicious_file_modifications.toml [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+commonly used for persistence on Linux systems. The rule detects modifications to files that are commonly used for
+cron jobs, systemd services, message-of-the-day (MOTD), SSH configurations, shell configurations, runtime control,
+init daemon, passwd/sudoers/shadow files, Systemd udevd, and XDG/KDE autostart entries. To leverage this rule, the
+paths specified in the query need to be added to the FIM policy in the Elastic Security app.
+"""
+from = "now-9m"
+index = ["logs-fim.event-*", "auditbeat-*"]
+language = "eql"
+license = "Elastic License v2"
+name = "Suspicious File Modification"
```
Comment (Samirbous) [inline rules/integrations/fim/persistence_suspicious_file_modifications.toml:20]: don't see anything suspicious (like unusual process), maybe `Potential Persistence via File Modification`

**Review by Samirbous — APPROVED** [review]

File: rules/integrations/fim/persistence_suspicious_file_modifications.toml (hunk 3 above)
Comment (Aegrah) [inline rules/integrations/fim/persistence_suspicious_file_modifications.toml:20]:
```suggestion
name = "Potential Persistence via File Modification"
```

**Review by Mikaayenson — APPROVED** [review]
LGTM

**Pull request merged**